libglusterfs/src/glusterfs/lkowner.h

```diff
@@ -19,8 +19,29 @@ lkowner_unparse(gf_lkowner_t *lkowner, char *buf, int buf_len)
 {
     int i = 0;
     int j = 0;
+    uint8_t nibble;
 
-    for (i = 0; i < lkowner->len; i++) {
+    /* the common case for lkowner is 8 bytes long
+       (a pointer) which will require 16 bytes (+ a NULL termination char)
+       long buffer to convert into.
+    */
+    if (lkowner->len >= 8 && buf_len > 16) {
```

Comment:
why not do this for the entire buffer, and not only the first bytes ?

Reply:
Because then I have the handle this part:
```
        if (i && !(i % 8)) {
            buf[j] = '-';
            j++;
        }
```